enhancements/network/baremetal-networking.md

```diff
@@ -0,0 +1,318 @@
+
+## Motivation
+
+### Goals
+
+Install an IPI OpenShift cluster on various on-premise non-cloud platforms that
+provides internal DNS and load balancing that is minimally required for OpenShift
+cluster to run properly. OpenShift installation will still require DHCP
+and DNS entries for api and the apps wildcard url.
+
+### Non-Goals
+
+## Proposal
+
+In cluster network infrastructure automates a number
+of components that are handled on other platforms by cloud infrastructure services.
+
+### Load-balanced control plane access
+
+Access to the Kubernetes API (port 6443) from clients both external
+and internal to the cluster should be load-balanced across control
+plane machines.
+
+Access to Ignition configs (port 22623) from clients within the
+cluster should also be load-balanced across control plane machines.
+
+### Virtual IP addresses and Keepalived
+
+[keepalived](https://keepalived.org) provides high-availablity for services by using a single
+"virtual" ip address (VIP) that can failover between multiple hosts using
+[VRRP](https://www.haproxy.com/documentation/hapee/1-5r2/configuration/vrrp/#understanding-vrrp)
+
+#### Bootstrap keepalived
+
+The bootstrap keepalived [pod](https://github.com/openshift/machine-config-operator/blob/master/manifests/baremetal/keepalived.yaml)
+is configured via the [keepalived.conf.tmpl](https://github.com/openshift/machine-config-operator/blob/master/manifests/baremetal/keepalived.conf.tmpl).
+
+The bootstrap keepalived VRRP instances have a higher weight using a `priority 50` than the control plane's `priority 40`.
+Bootstrap will maintain its role as master until some intervention which in our case is destroying the bootstrap node.
+
+#### Control Plane keepalived
+
+The control plane [pod](https://github.com/openshift/machine-config-operator/blob/master/templates/common/baremetal/files/baremetal-keepalived.yaml)
+is configured via the
+[baremetal-keepalived-keepalived.yaml](https://github.com/openshift/machine-config-operator/blob/master/templates/master/00-master/baremetal/files/baremetal-keepalived-keepalived.yaml)
+ignition template.
+
+The control plane keepalived configuration uses service checks to either add or remove points to the instance weight.
+Using the default `priority 40` and the service checks will determine which control plane node is the master VRRP instance.
+
+### API VIP (Virtual IP)
+
+In cluster network infrastructure, a VIP (Virtual IP) is used to provide
+failover of the API server across the control plane machines
+(including the bootstrap instance).
+
+The APIVIP is provided by the user
+via the `install-config.yaml` [parameter](https://github.com/openshift/installer/blob/master/pkg/types/baremetal/platform.go#L57-L63)
+or `openshift-installer` terminal prompts. The machine config operator does the inital
+render of the pod spec and configuration template.  The initContainer does the final
```

Comment:
> The machine config operator does the inital
render of the pod spec and configuration template.  The initContainer does the final
templating of the configuration with `baremetal-runtimecfg`.

let's keep a separate paragraph for `baremetal-runtimecfg`. Also can we provide details around what inputs does this take/provide as API...??